### ISSUE TITLE
Syntax Error in pgcna2?

### ISSUE THREAD

**Issue by mr-september** [medmaca/PGCNA#2]
Error:
```
 File "./pgcna2.py", line 141
    print("##----------------------------------------", str(datetime.now()), "----------------------------------------##", sep="")
                                                                                                                              ^
SyntaxError: invalid syntax
```
The `^` may be positioned differently for everyone, so to clarify, it's meant to point directly at the `=` between `sep` and `""`, all the way to the right.

This is also my first time using this package, and python overall in a long time, is this a problem that could be caused by improper setup or input data format? I've attached my #FileInfo.txt and the TSV expression matrix for you to double check.

[#FileInfo.txt](https://github.com/medmaca/PGCNA/files/6059522/FileInfo.txt)

[DEg_cluster_0.txt](https://github.com/medmaca/PGCNA/files/6059520/DEg_cluster_0.txt)

**Comment by medmaca** [comment]
Hi,

This looks to me like you're using py2 rather than the required py3 to run pgcna2. 

Look at https://github.com/medmaca/PGCNA/tree/master/PGCNA2 for more detailed requirements.

**Presuming you were wanting to find modules of genes:**

Regarding your files.  I've modified #FileInfo.txt so that it is tab separated and only includes 1 line of header.  I've modified your expression data by transposing it and removing zero expressed genes as this will cause errors (this was assuming you might try to run pgcna with -f 1; i.e. including all the data)

The new files:
[#FileInfo.txt](https://github.com/medmaca/PGCNA/files/6060471/FileInfo.txt)
[DEg_cluster_0_medmaca2.txt](https://github.com/medmaca/PGCNA/files/6060473/DEg_cluster_0_medmaca2.txt)

Also given the sparsity of the data I would suggest running using the **--usePearson** flag.

**Presuming you're trying to cluster the cells/groups**

For this you can use you're original DEg_cluster_0.txt, however, you'll need to modify my tab separated #FileInfo.txt to point to this file.  Also don't forget to run with the **-f 1** command if you're wanting to include all the data, else only the 80% most variant will be retained.

**NOTE**
PGCNA wasn't really designed for scRNA-seq analysis.  However, using the **--usePearson** flag it will produce results, but know that you might want to try methods that were specifically designed for such data.

**Comment by mr-september** [comment]
Thanks for the (very fast and helpful) reply!

I *should* be running this with Python 3.9 in PyCharm, on Ubuntu 20.10 (debian-based). Could there be other causes of this issue?

> Regarding your files. I've modified #FileInfo.txt so that it is tab separated and only includes 1 line of header. I've modified your expression data by transposing it and removing zero expressed genes as this will cause errors (this was assuming you might try to run pgcna with -f 1; i.e. including all the data)

Thank you! Realised I interpreted the readme wrongly, oops.

> Also don't forget to run with the -f 1 command

> PGCNA wasn't really designed for scRNA-seq analysis. However, using the --usePearson flag it will produce results

Again, thank you for the advice!

From your research, is there any particular method you would recommend for scRNA data? I mainly require a method that also reports anti-correlation (numerically -ve edges) relationships.

**Comment by medmaca** [comment]
if you write the script:
```
import sys
print(sys.version)
```
And run it in PyCharm, what do you see?

**Comment by mr-september** [comment]
```
3.9.0+ (default, Oct 19 2020, 09:51:18) 
[GCC 10.2.0]
```

**Comment by medmaca** [comment]
This is baffling to me.

This is what I see with different anaconda python versions (the pgcna2 help message is truncated in each case):

**python 2.7**

```
(base) C:\Users\Matthew\Desktop\DEMO>python
Python 2.7.16 |Anaconda, Inc.| (default, Mar 14 2019, 15:42:17) [MSC v.1500 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>>

(base) C:\Users\Matthew\Desktop\DEMO>python pgcna2.py -h
  File "pgcna2.py", line 141
    print("##----------------------------------------", str(datetime.now()), "----------------------------------------##", sep="")
                                                                                                                              ^
SyntaxError: invalid syntax
```

**python 3.8**
```
(base) C:\Users\Matthew\Desktop\DEMO>conda activate py38

(py38) C:\Users\Matthew\Desktop\DEMO>python
Python 3.8.3 (default, Jul  2 2020, 17:30:36) [MSC v.1916 64 bit (AMD64)] :: Anaconda, Inc. on win32
Type "help", "copyright", "credits" or "license" for more information.
>>>

(py38) C:\Users\Matthew\Desktop\DEMO>python pgcna2.py -h
usage: pgcna2.py [-h] [-w WORKFOLDER] [-d DATAF] [-m METAINF] [-s FILESEP] [-f RETAINF] [-g GENEFRAC] [-e EDGEPG] [-r ROUNDL] [--outF OUTF] [--corrMatF CORRMATF] [--corrMatFS CORRMATFS] [--gephiF GEPHIF] [--noLeidenalg] [--usePearson]
                 [--keepBigFA] [--keepBigF] [--ignoreDuplicates] [--singleCorr] [--singleCorrL] [--singleCorrListF SINGLECORRLISTF] [--corrChunk CORRCHUNK] [-n LARUNNUM] [-b LABESTPERC] [--lBaseFold LBASEFOLD] [--lClustTxtFold LCLUSTTXTFOLD]
                 [--lClustListFold LCLUSTLISTFOLD]

optional arguments:
  -h, --help            show this help message and exit
------------------------------------
```

**python 3.9**


```
(py38) C:\Users\Matthew\Desktop\DEMO>conda activate py39_base

(py39_base) C:\Users\Matthew\Desktop\DEMO>python
Python 3.9.2 | packaged by conda-forge | (default, Feb 21 2021, 04:59:43) [MSC v.1916 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>>

(py39_base) C:\Users\Matthew\Desktop\DEMO>python pgcna2.py -h
usage: pgcna2.py [-h] [-w WORKFOLDER] [-d DATAF] [-m METAINF] [-s FILESEP] [-f RETAINF] [-g GENEFRAC] [-e EDGEPG] [-r ROUNDL] [--outF OUTF] [--corrMatF CORRMATF] [--corrMatFS CORRMATFS] [--gephiF GEPHIF] [--noLeidenalg] [--usePearson]
                 [--keepBigFA] [--keepBigF] [--ignoreDuplicates] [--singleCorr] [--singleCorrL] [--singleCorrListF SINGLECORRLISTF] [--corrChunk CORRCHUNK] [-n LARUNNUM] [-b LABESTPERC] [--lBaseFold LBASEFOLD] [--lClustTxtFold LCLUSTTXTFOLD]
                 [--lClustListFold LCLUSTLISTFOLD]

optional arguments:
  -h, --help            show this help message and exit
 ----------------------------------------------
```

This works both on Windows and also Linux.

Can you try running pgcna2.py from the terminal in a similar fashion (i.e. show me entering python first).

**Comment by mr-september** [comment]
Figured out the issue - the "Terminal" in PyCharm is still running the system's default python 2.x, while the "Python Console" is python 3.x. Instead of running `./pgcna2.py -w <workFolderPath> -d <expressionFilesFolderPath>`, I had to run `python3 pgcna2.py -w <workFolderPath> -d <expressionFilesFolderPath>`.

I don't know if this is a good way to do things? Is there a way to make the Terminal automatically use PyCharm's interpreter?
